napari/layers/labels/_labels_mouse_bindings.py

```diff
@@ -16,7 +37,13 @@ def paint(layer, event):
             last_cursor_coord, layer.coordinates, layer.brush_size
         )
         for c in interp_coord:
-            layer.paint(c, layer.selected_label, refresh=False)
+            if layer._mode == Mode.PAINT:
+                layer.paint(c, layer.selected_label, refresh=False)
+            elif (
+                layer._mode == Mode.FILL
+                and layer.get_value() != layer.selected_label
+            ):
```

Comment:
```suggestion
            elif layer._mode == Mode.FILL:
```

I think we should drop the `and layer.get_value() != layer.selected_label` check here and put like that inside the `fill` function as I think the `get_value` could lag one interpolated coordinate behind the `c` coordinate, and we make two `get_value` calls instead of one. See comment below

Reply:
good call, added